### DISCUSSION TITLE
如何在 stencil load group 方法执行后自动计算 group 高度？

### DISCUSSION THREAD

**Discussion by ghost** [antvis/X6#2744] [Q&A] [answered]
我想在 load group 之后让加载进去的组件在 stencil 直接显示，但如果在初始化 stencil 时没有设置 stencilGraphWidth 和 stencilGraphHeight，group 的高度为 0，因此无法展示。

只有通过设置 graph 的 fitToContent 方法才能展示出来

https://user-images.githubusercontent.com/77133362/193745830-2e45defc-98ee-4335-a44b-a595a762d031.mp4

```js
// 这段代码在 node_modules\@antv\x6\src\addon\stencil\index.ts 文件中 
    Object.values(stencil.graphs).forEach((graph) => {
      graph.fitToContent({
        // minWidth: graph.options.width,
        // gridHeight: 1,
        // padding: 10
      });
    });
```
但是这个方法是 protected 的，不应该被外界调用。

所以想问一下有其他的方式去实现这个需求吗？或者把这个方法暴露出来，允许手动调用？

[![Edit vibrant-voice-xbpo3z](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/vibrant-voice-xbpo3z?fontsize=14&hidenavigation=1&theme=light)

**Comment by NewByVector** [comment, marked as answer]
可以设置 stencilGraphHeight: 0 来做到自适应。
